specifyweb/statistics/views.py

```diff
@@ -0,0 +1,126 @@
+
+from specifyweb.specify.views import login_maybe_required
+from specifyweb.specify.api import toJson
+from specifyweb.context import app_resource
+# from specifyweb.stored_queries.models import Determination, Taxon
+import logging
+
+logger = logging.getLogger(__name__)
+from django.db import connection
+
+
+@openapi(schema={
+    'get': {
+        'responses': {
+            '200': {
+                'description': 'Returns Global Collection Stats for Specify that backend delivers',
+                'content': {
+                    'application/json': {
+                        'schema': {
+                            'type': 'object',
+                            'properties': {
+                                'holdings': {
+                                    'type': 'object',
+                                    'properties': {
+                                        'familiesRepresented': {
+                                            'type': 'integer'
+                                        },
+                                        'generaRepresented': {
+                                            'type': 'integer'
+                                        },
+                                        'speciesRepresented': {
+                                            'type': 'integer'
+                                        }
+                                    }
+                                },
+                                'preparation': {
+                                    'type': 'object',
+                                    'additionalProperties': True,
+                                },
+                                'typeSpecimen': {
+                                    'type': 'object',
+                                    'additionalProperties': True,
+                                },
+                                'localityGeography': {
+                                    'type': 'object',
+                                    'properties': {
+                                        'countries': {
+                                            'type': 'integer'
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                },
+            }
+        }
+    }}, )
+def collection_global(request) -> http.HttpResponse:
+    cursor = connection.cursor()
+    # Holdings
+    # Families
+    holding_dict = {}
+    cursor.execute("""
+    SELECT count(*) FROM (SELECT DISTINCT tx.TaxonID FROM (SELECT DISTINCT tax.TaxonID,tax.nodenumber FROM (SELECT TaxonID FROM determination WHERE CollectionMemberID = %s AND determination.IsCurrent = true) as Families1,taxon as tax WHERE tax.isaccepted <> 0 and tax.TaxonID = Families1.TaxonID) as Families2, taxon as tx WHERE tx.rankid = 140 and Families2.nodenumber between tx.nodenumber and tx.HighestChildNodeNumber) as Familes3

```

Comment:
Lines should not be longer than 80 characters. Split this query into multiple lines to make it readable

Reply:
I like to break the queries according to the SQL clauses to show the structure. Like:
```
SELECT ...
FROM ...
JOIN ...
WHERE ...
AND ...
```

I try to format parenthesized expressions as indented blocks:
```
SELECT ...
FROM (
   SELECT ...
    FROM ...
)
```